panic: runtime error: makeslice: cap out of range

Getting a panic on promxy v0.0.56:

```
time="2019-12-02T17:08:23Z" level=info msg="promxy starting"
panic: runtime error: makeslice: cap out of range

goroutine 288 [running]:
github.com/jacksontj/promxy/vendor/github.com/prometheus/prometheus/promql.(*VectorSelector).String(0xc000e28000, 0xc000b24000, 0x26dc0e0)
	/go/src/github.com/jacksontj/promxy/vendor/github.com/prometheus/prometheus/promql/printer.go:180 +0x64
github.com/jacksontj/promxy/pkg/promclient.(*AddLabelClient).Query(0xc000aedd00, 0x2734200, 0xc000b24000, 0xc000dbe090, 0x21, 0x1bc69880, 0xed5773c33, 0x3937c20, 0x3, 0x3, ...)
	/go/src/github.com/jacksontj/promxy/pkg/promclient/label.go:114 +0x1b6
github.com/jacksontj/promxy/pkg/promclient.(*MultiAPI).Query.func1(0x2734200, 0xc000b24000, 0xc000e30e60, 0x0, 0xc00003a240, 0x2744c80, 0xc000aedd00, 0xc000dbe090, 0x21, 0x1bc69880, ...)
	/go/src/github.com/jacksontj/promxy/pkg/promclient/multi_api.go:289 +0xd2
created by github.com/jacksontj/promxy/pkg/promclient.(*MultiAPI).Query
	/go/src/github.com/jacksontj/promxy/pkg/promclient/multi_api.go:287 +0x300
```

--- a/pkg/promclient/labelfilter.go
+++ b/pkg/promclient/labelfilter.go
@@ -66,5 +66,17 @@ func FilterMatchers(ls model.LabelSet, matchers []*labels.Matcher) ([]*labels.Ma
 			filteredMatchers = append(filteredMatchers, matcher)
 		}
 	}
+
+	// Prometheus doesn't support empty matchers (https://github.com/prometheus/prometheus/issues/2162)
+	// so if we filter out all matchers we want to replace the empty matcher
+	// with a matcher that does the same
+	if len(filteredMatchers) == 0 {
+		filteredMatchers = append(filteredMatchers, &labels.Matcher{
+			Type:  labels.MatchRegexp,
+			Name:  labels.MetricName,
+			Value: ".+",
+		})
+	}
+
 	return filteredMatchers, true
 }

